catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImplTest.java

```diff
@@ -13,6 +13,1240 @@
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response);
+        when(parser.unmarshal(any(ParserConfigurator.class),
+                eq(JAXBElement.class),
+                any(InputStream.class))).thenReturn(jaxbRegistryPackage);
+        RegistryPackageType registryPackage =
+                federationAdminServiceImpl.getRegistryObjectByRegistryId(metacardId, null);
+        assertThat(registryPackage, notNullValue());
+        verify(federationAdminServiceImpl).getRegistryObjectByRegistryId(metacardId, null);
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void getRegistryObjectByRegistryIdEmptyString() throws FederationAdminException {
+        assertThat(federationAdminServiceImpl.getRegistryObjectByRegistryId(""), is(nullValue()));
+    }
+
+    @Test
+    public void getRegistryObjectByRegistryIdThatDoesNotExist() throws Exception {
+        QueryRequest request = getTestQueryRequest();
+        QueryResponse response = getPopulatedTestQueryResponse(request);
+        Metacard metacard = getTestMetacard();
+        when(security.getSystemSubject()).thenReturn(subject);
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response);
+        when(registryTransformer.transform(any(InputStream.class))).thenReturn(metacard);
+        assertThat(federationAdminServiceImpl.getRegistryObjectByRegistryId("Not a metacard"),
+                is(nullValue()));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void getRegistryObjectByRegistryIdDuplicateMetacards() throws Exception {
+        String metacardId = TEST_METACARD_ID;
+        Metacard metacard = testMetacard;
+        QueryRequest request = getTestQueryRequest();
+        QueryResponse response = getPopulatedTestQueryResponse(request, metacard, metacard);
+        when(security.getSystemSubject()).thenReturn(subject);
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response);
+        federationAdminServiceImpl.getRegistryObjectByRegistryId(metacardId);
+        verify(catalogFramework).query(any(QueryRequest.class));
+        verify(parser, never()).unmarshal(any(ParserConfigurator.class),
+                eq(JAXBElement.class),
+                any(InputStream.class));
+    }
+
+    @Test
+    public void initWithNoPreviousEntry() throws Exception {
+        QueryRequest request = getTestQueryRequest();
+        QueryResponse response = getPopulatedTestQueryResponse(request);
+        Metacard metacard = getTestMetacard();
+        when(security.getSystemSubject()).thenReturn(subject);
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response);
+        when(registryTransformer.transform(any(InputStream.class))).thenReturn(metacard);
+        federationAdminServiceImpl.init();
+        verify(federationAdminServiceImpl).addRegistryEntry(metacard);
```

Comment:
As mentioned above, we shouldn't be verifying that the class under test is calling one of its own methods. We need to either assert on the state of the `federationAdminServiceImpl` or use `verify` on the mocks used internally.

This comment applies to many of the test methods.

(This type of "loopback" testing is usually needed when the class under test has too many dependencies or is too large and needs to be broken up.)